Src/AutoFixture/ElementsBuilder.cs

```diff
@@ -0,0 +1,64 @@
+﻿using Ploeh.AutoFixture.Kernel;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Ploeh.AutoFixture
+{
+    /// <summary>
+    /// Draws a random element from the given collection
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public sealed class ElementsBuilder<T> : ISpecimenBuilder
+    {
+        private readonly T[] elements;
+        private readonly RandomNumericSequenceGenerator sequence;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ElementsBuilder{T}"/> class.
+        /// </summary>
+        /// <param name="elements">The collection from which elements should be drawn from.
+        /// It must contain at least one element.</param>
+        public ElementsBuilder(IEnumerable<T> elements)
+        {
+            if (elements == null)
+                throw new ArgumentNullException("elements");
+
+            this.elements = elements.ToArray();
+
+            if (this.elements.Length < 1)
+                throw new ArgumentException("elements must contain at least one element.", "elements");
```

Comment:
I'm almost embarrassed to mention this, because is such an incredibly insignificant issue, but I prefer that exception messages look like complete sentences. This one, I realise, *is* a complete sentence, because *elements* refer to the name of the argument, but due to the lower-case starting character, at first glance it *looks* incorrect.

Could we rephrase to something like the following?

> The supplied collection of elements must contain at least one element. This collection is expected to contain the elements from which the randomized algorithm will draw; if the collection is empty, there are no elements to draw.

If you don't care to make this change, I'm completely happy still accepting the pull request, but then I'll probably make the edit myself.